host/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM rust:1.75.0 as builder
+
+ENV DEBIAN_FRONTEND=noninteractive
+ARG BUILD_FLAGS=""
+
+# RUN apt-get update && apt-get install -y \
+#    build-essential \
+#    cmake \
+#    curl \
+#    git \
+#    libssl-dev \
+#    pkg-config \
+#    python3 \
+#    python3-pip \
+#    python3-setuptools \
+#    python3-wheel \
+#    ninja-build \
+#    wget
+# risc0 dependencies
+# RUN curl -L --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.sh | bash && \
+#    cargo binstall -y --force cargo-risczero && \
+#    cargo risczero build-toolchain && \
+#    cargo risczero install
+
+WORKDIR /opt/raiko
+COPY . .
+RUN cargo build --release --features "nitro" --features "docker_build"
+
+# Set up runtime
+FROM debian:bookworm-slim AS runner
+WORKDIR /raiko
+
+# Copy the Rust compiled binary from previous image
+COPY --from=builder /opt/raiko/target/release/raiko-host .
+
+CMD ["./raiko-host"]
```

Comment:
Right now will the full host run inside the enclave?

Reply:
yes.

Reply:
but TEE context will be enabled only for hash signing and report generation.

Reply:
We like to split the prover code from the program that actually needs to be proven in the "guest" to make sure we only prove the minimal code. This makes all provers work the same way (very important for zkVMs because there everything is very costsly) and it's nice to have it split up to know which parts are critical for security and which parts are not.

So i think for the nitro enclace proofs, similar to SGX, the prover would start the guest code program in the nitro enclave to generate the proof. Do you think it can work similarly?

Reply:
As discussed - initial implementation will be containerized with TEE context enter for signing and report generation only.